WordPress/Classes/Services/GravatarService.swift

```diff
@@ -1,37 +1,61 @@
 import Foundation
 
 
+@objc public enum GravatarServiceError: Int, Error {
+    case invalidAccountInfo
+}
+
 /// This Service exposes all of the valid operations we can execute, to interact with the Gravatar Service.
 ///
 open class GravatarService {
-    /// Designated Initializer
-    ///
-    /// - Parameter context: The Core Data context that should be used by the service.
+
+    /// This method fetches the Gravatar profile for the specified email address.
     ///
-    /// - Returns: nil if there's no valid WordPressCom Account available.
+    /// - Parameters:
+    ///     - email: The email address of the gravatar profile to fetch.
+    ///     - success: A success block.
+    ///     - failure: A failure block.
     ///
-    public init?(context: NSManagedObjectContext) {
-        let mainAccount = AccountService(managedObjectContext: context).defaultWordPressComAccount()
-        accountToken    = mainAccount?.authToken
-        accountEmail    = mainAccount?.email
-            .trimmingCharacters(in: CharacterSet.whitespaces)
-            .lowercased()
-
-        guard accountEmail?.isEmpty == false && accountToken?.isEmpty == false else {
-            return nil
-        }
+    open func fetchProfile(_ email: String, success:@escaping ((_ profile: GravatarProfile) -> Void), failure:@escaping ((_ error: NSError?) -> Void)) {
+        let remote = gravatarServiceRemote()
+        remote.fetchProfile(email, success: { (remoteProfile) in
+            let profile = GravatarProfile()
```

Comment:
I'd consider just returning the `RemoteGravatarProfile` here, but if you want to keep two separate types, how about giving `GravatarProfile` an initializer that takes a `RemoteGravatarProfile`?